core/src/banking_stage/vote_storage.rs

```diff
@@ -281,12 +291,59 @@ impl VoteStorage {
         // handled by the tpu vote thread
         matches!(self.vote_source, VoteSource::Gossip)
     }
+
+    /// Processes UNPROCESSED_BUFFER_STEP_SIZE packets at a time, returning the index of the first
+    /// packet that should be processed, the updated payload, and the packets that should be
+    /// processed.
+    pub fn march_iterator<'a, 'b>(
```

Comment:
again we don't need to have complicated iteration - see above comment. just literally iterate over the packets that we drain. read the fn documentation, it's already sorted in stake-weight order.